pyccl/massfunction.py

```diff
@@ -0,0 +1,68 @@
+from . import halos as hal
+from .pyutils import deprecated
+
+
+@deprecated(hal.MassFunc)
+def massfunc(cosmo, halo_mass, a, overdensity=200):
+    """Halo mass function, dn/dlog10M.
+    Args:
+        cosmo (:obj:`Cosmology`): Cosmological parameters.
+        halo_mass (float or array_like): Halo masses; Msun.
+        a (float): scale factor.
+        overdensity (float): overdensity parameter (default: 200)
+    Returns:
+        float or array_like: Halo mass function; dn/dlog10M.
+    """
+    mdef = hal.MassDef(overdensity, 'matter')
+    mf_par = cosmo._config_init_kwargs['mass_function']
+    if mf_par == 'tinker10':
+        mf = hal.MassFuncTinker10(cosmo, mdef)
+    elif mf_par == 'tinker':
+        mf = hal.MassFuncTinker08(cosmo, mdef)
+    elif mf_par == 'watson':
+        mf = hal.MassFuncWatson13(cosmo, mdef)
+    elif mf_par == 'shethtormen':
+        mf = hal.MassFuncSheth99(cosmo)
+    elif mf_par == 'angulo':
+        mf = hal.MassFuncAngulo12(cosmo)
+
+    return mf.get_mass_function(cosmo,
+                                halo_mass,
+                                a)
+
+
+@deprecated(hal.HaloBias)
+def halo_bias(cosmo, halo_mass, a, overdensity=200):
+    """Tinker et al. (2010) halo bias
+    Args:
+        cosmo (:obj:`Cosmology`): Cosmological parameters.
+        halo_mass (float or array_like): Halo masses; Msun.
+        a (float): Scale factor.
+        overdensity (float): Overdensity parameter (default: 200).
+    Returns:
+        float or array_like: Halo bias.
+    """
+    mdef = hal.MassDef(overdensity, 'matter')
+    mf_par = cosmo._config_init_kwargs['mass_function']
+    if mf_par == 'tinker10':
```

Comment:
Halo bias supports shethtormen too https://github.com/LSSTDESC/CCL/blob/v2.0.1/src/ccl_massfunc.c#L489

Reply:
Those bias functions are completely useless: they will return errors unless you use the virial overdensity **exactly**. I don't know anyone who ever managed to use them at all, because floating-point comparison of Delta_vir always failed.
I think it doesn't make sense to waste any cycles enforcing backward-compatibility of those.

Reply:
We have to for API compatibility. If someone, somewhere did manage to use them, we should support it. It can be as fragile as it was before, it just needs to be there.

Reply:
OK, but it won't be as fragile as it was before. This will now call the new implementation of this function, which will not check for Delta_vir (because it doesn't have to, this bias function was actually derived for FoF masses)

Reply:
That's fine. Then we have improved things! Improved implementations of APIs are great!

Reply:
:+1: 
OK, great.

Reply:
Done